Codec/Archive/Tar/Write.hs

```diff
@@ -70,7 +72,7 @@ putHeaderNoChkSum Entry {
     , putOct       8 $ permissions
     , putOct       8 $ ownerId ownership
     , putOct       8 $ groupId ownership
-    , putOct      12 $ contentSize
+    , numField    12 $ contentSize
```

Comment:
Hmm... we only need the long format if `contentSize` is out of range for `putOct`, no? This will also fix the TODO note in putOct afais.

And: we want this for all num fields afaiu.

> For fields containing numbers or timestamps that are out of range for the basic format, the GNU format uses a base-256 representation instead of an ASCII octal number. If the leading byte is 0xff (255), all the bytes of the field (including the leading byte) are concatenated in big-endian order, with the result being a negative number expressed in two’s complement form. If the leading byte is 0x80 (128), the non-leading bytes of the field are concatenated in big-endian order, with the result being a positive number expressed in binary form. Leading bytes other than 0xff, 0x80 and ASCII octal digits are reserved for future use, as are base-256 representations of values that would be in range for the basic format.

https://www.gnu.org/software/tar/manual/html_node/Extensions.html

Reply:
Yes probably `numField` should be used for other fields too, but I'm being conservative with this change.